src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            }
-        } else if (line.compare(JasmineGraphInstanceProtocol::AGGREGATE_CENTRALSTORE_TRIANGLES) == 0) {
-            instance_logger.log("Received : " + JasmineGraphInstanceProtocol::AGGREGATE_CENTRALSTORE_TRIANGLES, "info");
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphId = (data);
-            graphId = Utils::trim_copy(graphId, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID: " + graphId, "info");
+    bzero(data, INSTANCE_DATA_LENGTH + 1);
+    read(sockfd, data, INSTANCE_DATA_LENGTH);
+    string response = (data);
+    response = Utils::trim_copy(response, " \f\n\r\t\v");
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionId = (data);
-            partitionId = Utils::trim_copy(partitionId, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionId, "info");
+    if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE) == 0) {
+        std::cout << "Sending file " << filePath << " through port " << dataPort << std::endl;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionIdList = (data);
-            partitionIdList = Utils::trim_copy(partitionIdList, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID List : " + partitionIdList, "info");
+        FILE *fp = fopen(filePath.c_str(), "r");
+        if (fp == NULL) {
+            instance_logger.log("Error opening file", "error");
+            close(sockfd);
+            return false;
+        }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string priority = (data);
-            priority = Utils::trim_copy(priority, " \f\n\r\t\v");
-            instance_logger.log("Received priority: " + priority, "info");
+        for (;;) {
+            unsigned char buff[INSTANCE_FILE_BUFFER_LENGTH] = {0};
+            int nread = fread(buff, 1, INSTANCE_FILE_BUFFER_LENGTH, fp);
+            instance_logger.log("Bytes read " + to_string(nread), "info");
 
-            int threadPriority = std::atoi(priority.c_str());
+            /* If read was success, send data. */
+            if (nread > 0) {
+                instance_logger.log("Sending", "info");
```

Comment:
Lets remove this logger if we cannot find better log statement. Logging just one word does not make sense.

Reply:
This is replaced by the wrapper function in next PR

Reply:
Ok